src/OAuthLibrary/OAuthControl/OAuthWebBrowser.cs

```diff
@@ -171,6 +186,18 @@ public void BeginImplicitGrant()
             // Clear existing token
             AccessToken = "";
 
+            if (!string.IsNullOrEmpty(Org) && !string.IsNullOrEmpty(Provider) && !string.IsNullOrEmpty(State))
```

Comment:
This is ok but it would be cleaner to start off with a variable like
string URL = $"https:\\\\login.{Environment}/authorize?client_id={ClientId}&response_type=token&redirect_uri={RedirectUri}";

then add `&state={State}` onto it if state is set and do the same thing for provider and org.
This might be a good use for StringBuilder: https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-6.0

Reply:
You could just do
```
if (!string.IsNullOrEmpty(Org))
{
    sb.Append($"&org={Org}");
}
if (!string.IsNullOrEmpty(Provider))
{
    sb.Append($"&provider={Provider}");
}
if (!string.IsNullOrEmpty(State))
{
    sb.Append($"&state={State}");
}
```